src/Models/ShallowWaterModels/solution_and_tracer_tendencies.jl

```diff
@@ -7,34 +7,34 @@ using Oceananigans.TurbulenceClosures: ∇_dot_qᶜ, ∂ⱼ_τ₁ⱼ, ∂ⱼ_τ
 @inline half_g_h²(i, j, k, grid, h, g)  = @inbounds 1/2 * g * h[i, j, k]^2
 @inline h_plus_hB(i, j, k, grid, h, hB) = @inbounds h[i, j, k] + hB[i, j, k]
 
-@inline x_pressure_gradient(i, j, k, grid, g, h, hB, ::SingleLayerModel, formulation) = ∂xᶠᶜᶜ(i, j, k, grid, half_g_h², h, g)
-@inline y_pressure_gradient(i, j, k, grid, g, h, hB, ::SingleLayerModel, formulation) = ∂yᶜᶠᶜ(i, j, k, grid, half_g_h², h, g)
+@inline x_pressure_gradient(i, j, k, grid, g, h, hB, ::SingleLayerModel, ::ConservativeFormulation) = ∂xᶠᶜᶜ(i, j, k, grid, half_g_h², h, g)
+@inline y_pressure_gradient(i, j, k, grid, g, h, hB, ::SingleLayerModel, ::ConservativeFormulation) = ∂yᶜᶠᶜ(i, j, k, grid, half_g_h², h, g)
     
-@inline function x_pressure_gradient(i, j, k, grid, g, h, hB, ::MultiLayerModel, formulation)
+@inline function x_pressure_gradient(i, j, k, grid, g, h, hB, ::MultiLayerModel, ::ConservativeFormulation)
     Nz = grid.Nz
     ## Should be set to grid dimensions
-    η[i, j, k] = hB[i, j, k] + cumsum(h, dims = 3)
+    η[i, j, k] = hB[i, j, k] + cumsum(h, dims = 3)[i, j, k]
```

Comment:
this `cumsum` might be very slow for a large number of layers. A way around it would be to add another field `∫h` to the model and precompute the sum

Reply:
True. I can work on it!